jaxrs-common/src/main/java/io/micronaut/jaxrs/common/JaxRsMessageBodyReader.java

```diff
@@ -41,34 +44,57 @@
 @EachBean(jakarta.ws.rs.ext.MessageBodyReader.class)
 public final class JaxRsMessageBodyReader<T> implements MessageBodyReader<T> {
 
-    private final jakarta.ws.rs.ext.MessageBodyReader<T> messageBodyReader;
+    private final jakarta.ws.rs.ext.MessageBodyReader<T> delegate;
+    private final List<ReaderInterceptor> readerInterceptor;
 
-    JaxRsMessageBodyReader(jakarta.ws.rs.ext.MessageBodyReader<T> messageBodyReader) {
-        this.messageBodyReader = messageBodyReader;
+    public JaxRsMessageBodyReader(jakarta.ws.rs.ext.MessageBodyReader<T> delegate,
+                                  List<ReaderInterceptor> readerInterceptor) {
+        this.delegate = delegate;
+        this.readerInterceptor = readerInterceptor;
+        JaxRsUtils.sortByPriority(readerInterceptor);
     }
 
-    @Override
-    public boolean isReadable(@NonNull Argument<T> type, @Nullable MediaType mediaType) {
-        return messageBodyReader.isReadable(type.getType(), type.asType(), type.getAnnotationMetadata().synthesizeAll(), as(mediaType));
+    public jakarta.ws.rs.ext.MessageBodyReader<T> getDelegate() {
+        return delegate;
     }
 
-    private jakarta.ws.rs.core.MediaType as(MediaType mediaType) {
-        return jakarta.ws.rs.core.MediaType.valueOf(mediaType.toString());
+    @Override
+    public boolean isReadable(@NonNull Argument<T> type, @Nullable MediaType mediaType) {
+        return delegate.isReadable(type.getType(), type.asType(), type.getAnnotationMetadata().synthesizeAll(), JaxRsUtils.convert(mediaType));
     }
 
     @Override
     public @Nullable T read(@NonNull Argument<T> type, @Nullable MediaType mediaType, @NonNull Headers httpHeaders, @NonNull InputStream inputStream) throws CodecException {
         try {
-            return messageBodyReader.readFrom(
+            Iterator<ReaderInterceptor> iterator = readerInterceptor.iterator();
+            if (iterator.hasNext()) {
```

Comment:
only one supported?

Reply:
The next is triggered in the interceptor context